app/database/item_store.go

```diff
@@ -309,119 +297,37 @@ func (s *ItemStore) checkSubmissionRightsForTimeLimitedContest(itemID int64, use
 			"of a different competition than the one in progress")
 }
 
-type activeContestInfo struct {
-	ItemID                   int64
-	UserID                   int64
-	ContestEnteringCondition string
-	IsTeamContest            bool
-
-	Now               time.Time
-	DurationInSeconds int32
-	EndTime           time.Time
-	StartTime         time.Time
-}
-
-func (contest *activeContestInfo) IsOver() bool {
-	return contest.EndTime.Before(contest.Now) || contest.EndTime.Equal(contest.Now)
-}
-
-// Closes the time-limited contest if needed or returns time stats
-func (s *ItemStore) getActiveContestInfoForUser(user *User) *activeContestInfo {
-	// Get info for the item if the user has already started it, but hasn't finished yet
+func (s *ItemStore) getActiveContestItemIDForUser(user *User) *int64 {
+	// Get id of the item if the user has already started it, but hasn't finished yet
 	// Note: the current API doesn't allow users to have more than one active contest
 	// Note: groups_attempts rows with 'entered_at' should exist to make this function return the info
-	var results []struct {
-		Now                      Time
-		DurationInSeconds        int32
-		ItemID                   int64
-		AdditionalTimeInSeconds  int32
-		EnteredAt                Time
-		ContestEnteringCondition string
-		IsTeamContest            bool
-	}
-	mustNotBeError(s.
-		Select(`
-			NOW() AS now,
-			TIME_TO_SEC(items.duration) AS duration_in_seconds,
-			items.id AS item_id,
-			items.contest_entering_condition,
-			items.has_attempts AS is_team_contest,
-			IFNULL(SUM(TIME_TO_SEC(groups_contest_items.additional_time)), 0) AS additional_time_in_seconds,
-			MIN(contest_participations.entered_at) AS entered_at,
-			MIN(contest_participations.finished_at) AS finished_at`).
+	var itemID int64
+
+	err := s.
+		Select("items.id AS item_id").
 		Joins(`
 			JOIN groups_ancestors_active
 				ON groups_ancestors_active.child_group_id = ?`, user.GroupID).
-		Joins(`LEFT JOIN groups_attempts AS contest_participations ON contest_participations.item_id = items.id AND
+		Joins(`
+			JOIN groups_groups_active
+				ON groups_groups_active.parent_group_id = items.contest_participants_group_id AND
+					groups_groups_active.child_group_id = groups_ancestors_active.child_group_id`).
+		Joins(`JOIN groups_attempts AS contest_participations ON contest_participations.item_id = items.id AND
 			contest_participations.group_id = groups_ancestors_active.ancestor_group_id AND
 			contest_participations.entered_at IS NOT NULL`).
 		Joins(`
 			LEFT JOIN groups_contest_items ON groups_contest_items.item_id = items.id AND
 				groups_contest_items.group_id = groups_ancestors_active.ancestor_group_id`).
 		Group("items.id").
 		Order("MIN(contest_participations.entered_at) DESC").
-		Having("entered_at IS NOT NULL AND finished_at IS NULL").
-		Limit(1).Scan(&results).Error())
+		PluckFirst("items.id", &itemID).Error()
```

Comment:
I don't get this request, you join with `groups_contest_items` for instance which is not used in the result of the request. groups_attempts are not used as well (or for filtering?).

Reply:
fixed